src/ParquetMsg.chpl

```diff
@@ -983,57 +985,20 @@ module ParquetMsg {
           }
           rnames.pushBack((dsetname, ObjType.PDARRAY, valName));
         } else if ty == ArrowTypes.stringArr {
-          /*
-            1. create a block distributed files array (locale owner reads file)
-            2. get number of row groups so we know how much data we have to store
-            3. create array to store data (2D array with same distribution dist files)
-            4. go distributed and create readers for each file
-          */
-          extern proc c_getNumRowGroups(readerIdx): c_int;
-          extern proc c_openFile(filename, idx);
-          extern proc c_createRowGroupReader(rowGroup, readerIdx);
-          extern proc c_createColumnReader(colname, readerIdx);
-          extern proc c_freeMapValues(rowToFree);
-          extern proc c_readParquetColumnChunks(filename, batchSize,
-                                                numElems, readerIdx, numRead,
-                                                externalData, defLevels, errMsg): int;
-
           var entrySeg = createSymEntry(len, int);
 
-          var distFiles = makeDistArray(filenames);
-          var numRowGroups: [distFiles.domain] int;
-
-          var maxRowGroups = getRowGroupNums(distFiles, numRowGroups);
-          var externalData: [distFiles.domain] [0..#maxRowGroups] c_ptr(void);
-          var containsNulls: [distFiles.domain] [0..#maxRowGroups] bool;
-          var valsRead: [distFiles.domain] [0..#maxRowGroups] int;
-          var bytesPerRG: [distFiles.domain] [0..#maxRowGroups] int;
-          var startIdxs: [distFiles.domain] [0..#maxRowGroups] int; // correspond to starting idx in entrySeg
-
-          fillSegmentsAndPersistData(distFiles, entrySeg, externalData, containsNulls, valsRead, dsetname, sizes, len, numRowGroups, bytesPerRG, startIdxs);
-
-          var (rgSubdomains, totalBytes) = getRGSubdomains(bytesPerRG, maxRowGroups);
-          
-          var entryVal;
-          
-          if containsNulls[0][0] {
+          // Calculate byte sizes by reading or fixed length
+          if fixedLen < 2 {
             byteSizes = calcStrSizesAndOffset(entrySeg.a, filenames, sizes, dsetname);
             entrySeg.a = (+ scan entrySeg.a) - entrySeg.a;
-            entryVal = createSymEntry((+ reduce byteSizes), uint(8));
-            readStrFilesByName(entryVal.a, whereNull, filenames, byteSizes, dsetname, ty);
           } else {
-            entryVal = createSymEntry(totalBytes, uint(8));
-            entrySeg.a = (+ scan entrySeg.a) - entrySeg.a;
-            copyValuesFromC(entryVal, distFiles, externalData, valsRead, numRowGroups, rgSubdomains, maxRowGroups, sizes, entrySeg.a, startIdxs);
+            entrySeg.a = fixedLen;
```

Comment:
don't we still need to do something like `entrySeg.a = (+ scan entrySeg.a) - entrySeg.a;`? Cause to me this reads as the lengths not the segs (so every val in `entrySeg.a` would be like 5 or whatever)

Reply:
Whoops, yup, you got me. I was experimenting with some performance stuff and guess that slipped by. For context, this is what I was seeing for performance:

regular
| nl | Single file per locale | 10 files per locale |
| -- | ----------------------:| -------------------:|
| 1  |                0.13s   |            1.91s    |
| 16 |                1.00s   |            16.0s    |

fixed length
| nl | Single file per locale | 10 files per locale |
| -- | ----------------------:| -------------------:|
| 1  |                0.08s   |            1.35s    |
| 16 |                0.95s   |            23.9s    |

So, even though we are doing strictly less work with the fixed length approach, the performacne is still worse, I suspect due to some file caching problems, but I am still working to pin down the root of the issue there.

What that means is that the fixed length arg doesn't result in a perf improvement all the time, but having it in place still allows you to get it in the cases where it does (but really, will help make my performance experiments easier since I can make more isolated changes 😄)